docs/audit-log-filter/audit-log-filter-options.md

```diff
@@ -0,0 +1,419 @@
+
+Defines pruning based on the combined size of the files:
+
+* 0 (zero), which is the default value, disables pruning based on size.
+* A value greater than 0 (zero) enables pruning based on size and defines the limit of the combined size. When the files exceed this limit, they can be pruned.
+
+The value is based on 4096 (block size). A value is truncated to the nearest multiple of the block size. If the value is less than 4096, the value is treated as 0 (zero).
+
+If the values for `audit_log_filter_rotate_on_size` and `audit_log_filter_max_size` are greater than 0, we recommend that `audit_log_filter_max_size` value should be at least seven times the `audit_log_filter_rotate_on_size` value. A lower value generates a warning to the error log and may remove all of the rotated logs each time.
+
+Pruning requires the following options:
+
+* `audit_log_filter_max_size`
+* `audit_log_filter_rotate_on_size`
+* `audit_log_filter_prune_seconds`
+
+### audit_log_filter_password_history_keep_days
+
+| Option name | Description |
+| --- | --- |
+| Command-line | --audit-log-filter-password-history-keep-days |
+| Dynamic | Yes  |
+| Scope | Global  |
+| Data type | Integer  |
+| Default | 0  |
+
+Defines when passwords that are archived and is measured in days.
+
+Encrypted log files have passwords stored in the keyring. The plugin also stores a password history. A password does not expire, despite being past the value, in case the password is used for rotated audit logs.The operation of creating a password also archives the previous password.
+
+The default value is 0 (zero). This value disables the expiration of passwords. Passwords are retained forever. 
+
+If the plugin starts and if encryption is enabled, the plugin checks for an audit log filter encryption password. If a password is not found, the plugin generates a random password.
+
+Call `audit_log_filter_encryption_set() to set a specific password.
+
+### audit_log_filter_prune_seconds
+
+| Option name | Description |
+| --- | --- |
+| Command-line | --audit-log-filter-prune-seconds |
+| Dynamic | Yes  |
+| Scope | Global  |
+| Data type | Integer  |
+| Default | 0  |
+| Minimum value | 0 |
+| Maximum value | 1844674073709551615 |
+| Unit | bytes |
+
+This option is only supported for JSON-format files.
+
+Defines when the audit log filter file is pruned. This pruning is based on the age of the file. The value is measured in seconds. The plugin supports pruning for JSON format only.
+
+A value of 0 (zero) is the default and disables pruning. The maximum value is 18446744073709551615.
+
+A value greater than 0 enables pruning. An audit log filter file can be pruned after this value.
+
+A JSON log pruning operation is based on the values of [`audit_log_filter_rotate_on_size`](#audit_log_filter_rotate_on_size), [`audit_log_filter_max_size`](#audit_log_filter_max_size) and [`audit_log_filter_prune_seconds`](#audit_log_filter_prune_seconds). Setting only one of these values does not cause pruning.
```

Comment:
"Setting only one of these values does not cause pruning"

This may be a bit misleading. To enable log pruning you need to:
- enable log rotation by setting audit_log_filter_rotate_on_size;
- set one of audit_log_filter_max_size or audit_log_filter_prune_seconds to choose criteria for pruned files selection.

Reply:
rewritten